emission/storage/decorations/trip_queries.py

```diff
@@ -146,7 +170,9 @@ def final_candidate(filter_fn, potential_candidates):
     # input before the pipeline is run, and then overwrites after pipeline is
     # run
     sorted_pc = sorted(extra_filtered_potential_candidates, key=lambda c:c["metadata"]["write_ts"])
-    entry_detail = lambda c: c.data.label if "label" in c.data else c.data.start_fmt_time
+    entry_detail = lambda c: getattr(c.data, "label", \
+                                getattr(c.data, "start_fmt_time", \
+                                getattr(c.data, "enter_fmt_time", None)))
```

Comment:
please add comment on why you use `getattr` (I assume to avoid AttributeNotFound exceptions)

Reply:
I'll refactor into helper functions 'begin' and 'stop'

(and I'll use `in` keyword instead of `getattr`)